spmd/tensor/ops/tensor_ops.py

```diff
@@ -53,3 +54,35 @@ def no_shard_prop_rule(op_schema: OpSchema) -> OutputSharding:
 
 for op in no_shard_prop_ops:
     DTensor._op_to_rules[op] = no_shard_prop_rule
+
+
+@register_prop_rule("aten.bucketize.Tensor")
+def prop_bucketize(op_schema: OpSchema) -> OutputSharding:
+    """
+    Point-wise on the first input. Expect replicated for second input.
```

Comment:
seems like we just forward the first input sharding instead of pointwise?

Reply:
right what i mean is that since it's point-wise on the first input, we can simply forward the sharding.

Reply:
let me improve the comment.